tests/test_spiral.py

```diff
@@ -0,0 +1,512 @@
+import numpy as np
+import galsim
+from galsim_test_helpers import (
+    check_basic,
+    do_shoot,
+    do_kvalue,
+    check_pickle,
+)
+from galsim.errors import (
+    GalSimValueError,
+    GalSimRangeError,
+    # GalSimIncompatibleValuesError,
+)
+import pytest
+
+
+def _get_example_kw():
+    return {
+        'npoints': 100,
+        'half_light_radius': 1.0,
+        'flux': 3.5,
+        'narms': 3,
+        'angle_fuzz': 0.07,
+        'xy_fuzz': 0.09,
+        'rel_height': 0.13,
+        'inclination': 45 * galsim.degrees,
+        'rotation': -20 * galsim.degrees,
+    }
+
+
+def test_spiral_defaults():
+    """
+    Create a spiral galaxy and test that the getters work for
+    default inputs
+    """
+
+    seed = 1234
+    npoints = 100
+    hlr = 1.0
+    rng = galsim.BaseDeviate(seed)
+    sp = galsim.Spiral(npoints, half_light_radius=hlr, rng=rng)
+
+    assert sp.npoints == npoints, (
+        'expected npoints==%d, got %d' % (npoints, sp.npoints)
+    )
+    assert sp.input_half_light_radius == hlr, (
+        'expected hlr==%g, got %g' % (hlr, sp.input_half_light_radius)
+    )
+
+    nobj = sp.points.shape[0]
+    assert nobj == npoints, (
+        'expected %d objects, got %d' % (npoints, nobj)
+    )
+
+    pts = sp.points
+    assert pts.shape == (npoints, 3), (
+        'expected (%d,2) shape for points, got %s' % (npoints, pts.shape)
+    )
+    np.testing.assert_almost_equal(sp.centroid.x, pts[:, 0].mean())
+    np.testing.assert_almost_equal(sp.centroid.y, pts[:, 1].mean())
+
+    sp.calculateHLR()
+    sp.calculate_e1e2()
```

Comment:
Should probably assert something about the return values of these two, rather than just that they run without error.